### PR TITLE
Operator support

### PR DESCRIPTION
**Pull request by petewall**
Adding support for operators by specifying the operator_dir and passing that along, just like the helm-chart-dir.

Also bringing in some unit tests for a function that I didn't end up using.  But, hey, free unit tests.

### CODE REVIEW

**Comment by petewall** [comment]
I have some final checks to do before this should be merged.
Reviews are fine, but no merges for now.

**Comment by petewall** [comment]
OK.  This has been tested and things look good to me.

File: tile_generator/package_flags.py [hunk 1]
```diff
@@ -18,12 +18,14 @@ def _update_compilation_vm_disk_size(manifest):
     return package_size
 
 
-def get_disk_size_for_chart(start_path):
+def get_disk_size_for_chart(*chart_paths):
```
Comment (keshto) [inline tile_generator/package_flags.py:21]: *chart_paths???

File: tile_generator/package_flags.py [hunk 2]
```diff
@@ -18,12 +18,14 @@ def _update_compilation_vm_disk_size(manifest):
...
     total_size = 0
-    for dirpath, dirnames, filenames in os.walk(start_path):
-        for f in filenames:
-            fp = os.path.join(dirpath, f)
-            total_size += os.path.getsize(fp)
+    for chart_path in chart_paths:
+        if chart_path != None:
```
Comment (keshto) [inline tile_generator/package_flags.py:24]: Will chart_path ever == None at this point?

File: tile_generator/package_flags.py [hunk 3]
```diff
@@ -485,7 +487,7 @@ def _apply(self, config_obj, package, release):
                     'name': 'kibosh',
                     'dynamic_ip': 1,
                     'varname': 'kibosh',
-                    'ephemeral_disk': package.get('ephemeral_disk', get_disk_size_for_chart(package['helm_chart_dir'])),
+                    'ephemeral_disk': package.get('ephemeral_disk', get_disk_size_for_chart(package['helm_chart_dir'], package['operator_dir'] if 'operator_dir' in package else None)),
```
Comment (keshto) [inline tile_generator/package_flags.py]: I don't think `else None` is needed here

Comment (keshto) [reply tile_generator/package_flags.py]: maybe just `package.get('operator_dir')`

File: tile_generator/package_flags.py [hunk 4]
```diff
@@ -503,14 +505,15 @@ def _apply(self, config_obj, package, release):
                             'token': '(( .properties.k8s_cluster_token.value ))',
                             'service_id': '(( .properties.service_id.value ))',
                             'helm_chart_dir': '/var/vcap/packages/charts_for_%s/chart' % packagename,
+                            'operator_dir': '/var/vcap/packages/charts_for_%s/operator_chart' % packagename,
```
Comment (keshto) [inline tile_generator/package_flags.py:508]: You seem to be repurposing/piggybacking the charts_for_%s directory. Did you ensure there is no adverse consequence of that? Does it make sense to have a operator_chart_for_%s?

File: tile_generator/package_flags.py [hunk 5]
```diff
@@ -503,14 +505,15 @@ def _apply(self, config_obj, package, release):
...
                         }
                     },
                 }, {
                     'name': 'loader',
                     'dynamic_ip': 1,
                     'post_deploy': True,
                     'lifecycle': 'errand',
-                    'ephemeral_disk': package.get('ephemeral_disk', get_disk_size_for_chart(package['helm_chart_dir']) * 3),
+                    'ephemeral_disk': package.get('ephemeral_disk', get_disk_size_for_chart(package['helm_chart_dir'], package['operator_dir'] if 'operator_dir' in package else None) * 3),
```
Comment (keshto) [inline tile_generator/package_flags.py]: See comment above about pythonic way of using `.get('operator_dir')`

**Review by keshto — CHANGES_REQUESTED** [review]
More unit tests woot woot!

File: tile_generator/package_flags.py (hunk 1 above)
Comment (petewall) [reply tile_generator/package_flags.py:21]: `*chart_paths` means that you can call this method with one or more parameters, so all of these are valid
`get_disk_size_for_chart(path)`
`get_disk_size_for_chart(path1, path2)`

`chart_paths` ends up being a list

File: tile_generator/package_flags.py (hunk 3 above)
Comment (petewall) [reply tile_generator/package_flags.py]: Yes, much nicer.  Will change to package.get()

File: tile_generator/package_flags.py (hunk 2 above)
Comment (petewall) [reply tile_generator/package_flags.py:24]: Yes.  If there is no operator_dir in the package structure, line 490 will send in a None.

File: tile_generator/package_flags.py (hunk 5 above)
Comment (petewall) [reply tile_generator/package_flags.py]: Done.

File: tile_generator/package_flags.py (hunk 4 above)
Comment (petewall) [reply tile_generator/package_flags.py:508]: The benefit for doubling-up in the charts_for_%s directory is that it allows for both directories to be handled in the loader errand.

After reviewing the code, there's a few changes that were missed, which only affected private repositories.  It's an additional parameter being passed to the loader errand definition and a change in the kibosh repository.

**Review by keshto — APPROVED** [review]

**Pull request merged**